irksome/dirk_stepper.py

```diff
@@ -142,12 +91,33 @@ def __init__(self, F, butcher_tableau, t, dt, u0, bcs=None,
         self.num_linear_iterations = 0
 
         self.butcher_tableau = butcher_tableau
+        self.num_stages = num_stages = butcher_tableau.num_stages
+        self.AAb = numpy.vstack((butcher_tableau.A, butcher_tableau.b))
+        self.CCone = numpy.append(butcher_tableau.c, 1.0)
+
+        # Need to be able to set BCs for either the DIRK or explicit cases.
+
+        # For DIRK, we say that the stage i solution should match the
```

Comment:
Is it the case that this is sufficiently accurate?  I guess we don’t have an implicit method, so this might be as good as we can get?

Reply:
For the stage values, there's 2 ways to impose BCs via an index-1 DAE, and they both agree that this is the right thing to do.  For DIRK, they would differ in the non-stiffly-accurate casein terms of whether we discard the reconstruction of the BCs that you get from the Butcher Tableau in favour of imposing the BCs at the next timestep.  I guess we could add an option to do that (just in case any non-stiffly-accurate DIRK aficionados are lurking out there), but it didn't seem like a great option to me.